code/modules/mob/living/carbon/inventory.dm

```diff
@@ -155,3 +155,35 @@
 /mob/living/carbon/proc/get_holding_bodypart_of_item(obj/item/I)
 	var/index = get_held_index_of_item(I)
 	return index && hand_bodyparts[index]
+
+/mob/living/carbon/proc/give()
+	var/obj/item/receiving = get_active_held_item()
+	if(!receiving)
+		return
+	visible_message("<span class='notice'>[src] is offering [receiving]</span>", "<span class='notice'You offer [receiving]</span>", null, 2)
+	for(var/mob/living/carbon/C in orange(1, src))
+		var/obj/screen/alert/give/G = C.throw_alert("Give", /obj/screen/alert/give)

```

Comment:
only one person can be offering like this which is meh, maybe make use something else as the category so you can get multiple givers.

Reply:
just checked; it doesn't care so you could just throw the ref to the giver in there like a boss.